protocol/x/listing/genesis.go

```diff
@@ -8,9 +8,15 @@ import (
 
 // InitGenesis initializes the module's state from a provided genesis state.
 func InitGenesis(ctx sdk.Context, k keeper.Keeper, genState types.GenesisState) {
+	k.InitializeForGenesis(ctx)
+
+	if err := k.SetMarketsHardCap(ctx, 0); err != nil {
```

Comment:
Why isn't this read from the genesis state?

Reply:
good catch